static/app/views/releases/thresholdsList/thresholdGroupRows.tsx

```diff
@@ -213,56 +206,45 @@ export function ThresholdGroupRows({
   return (
     <StyledThresholdGroup>
       {Array.from(thresholdIdSet).map((tId: string, idx: number) => {
-        const threshold = editingThresholds[tId] || thresholdsById[tId];
-        let environmentName;
-        if (editingThresholds[tId]) {
-          // editing environment type is String
-          environmentName = editingThresholds[tId].environment;
+        let threshold;
+        let isEditing = false;
+        if (tId in editingThresholds) {
+          isEditing = true;
+          threshold = editingThresholds[tId] as EditingThreshold;
         } else {
-          // Threshold environment type is Environnment
-          environmentName = thresholdsById[tId].environment
-            ? thresholdsById[tId].environment.name
-            : '';
+          threshold = initialThreshold as Threshold;
         }
+
         return (
           <StyledRow
             key={threshold.id}
-            lastRow={idx === thresholdIdSet.size - 1}
-            hasError={threshold.hasError}
+            lastRow={isLastRow && idx === thresholdIdSet.size - 1}
+            hasError={isEditing && (threshold as EditingThreshold).hasError}
           >
-            <FlexCenter style={{borderBottom: 0}}>
-              {idx === 0 ? (
-                <ProjectBadge
-                  project={threshold.project}
-                  avatarSize={16}
-                  hideOverflow
-                  disableLink
-                />
-              ) : (
-                ''
-              )}
-            </FlexCenter>
-            {/* TODO: IF this is a new threshold - allow environment select */}
-            {newGroup ? (
+            {/* ENV ONLY EDITABLE IF NEW */}
+            {!initialThreshold || threshold.id !== initialThreshold.id ? (
```

Comment:
why is this the case? for my own understanding

Reply:
Yea - because it doesn't make sense to move threshlods between environment.